pgactivity/types.py

```diff
@@ -607,20 +611,42 @@ class Host:
 
 
 @attr.s(auto_attribs=True, slots=True)
-class DBInfo:
-    total_size: int
-    size_ev: int
+class SwapInfo:
+    used: int
+    free: int
+    total: int
+    pct_used: Pct = attr.ib(init=False)
+    pct_free: Pct = attr.ib(init=False)
+
+    @classmethod
+    def default(cls) -> "SwapInfo":
+        return cls(0, 0, 0)
+
+    def __attrs_post_init__(self) -> None:
+        # prevents Frozen=True
+        self.pct_used = self.used * 100 / self.total
+        self.pct_free = self.free * 100 / self.total
 
 
 @attr.s(auto_attribs=True, slots=True)
 class MemoryInfo:
-    percent: float
     used: int
+    buff_cached: int
+    free: int
     total: int
+    pct_used: Pct = attr.ib(init=False)
+    pct_bc: Pct = attr.ib(init=False)
+    pct_free: Pct = attr.ib(init=False)
 
     @classmethod
     def default(cls) -> "MemoryInfo":
-        return cls(0.0, 0, 0)
+        return cls(0, 0, 0, 0)
+
+    def __attrs_post_init__(self) -> None:
+        # prevents Frozen=True
+        self.pct_used = self.used * 100 / self.total
```

Comment:
I moved the percent calculus here instead of in the `mem_swap_load()` function (in activities.py).

Is it the proper way to d it ?

Reply:
Since these percentage values are computed, they should be made available as properties of the class (where the computation will always be perform). This was, we're certain that the percentage value is always consistent with underlying data.

Reply:
Ok I'll rewrite it with __setattr__ to make the class frozen again.

Reply:
No, not `__setattr__`, this is equivalent to setting an attribute.
Rather:
```python
@property
def pct_used(self) -> float:
     return self.used * 100 / self.total
```

Reply:
Ok I understand now with the explanation above. thx.